andela/rivendell-ah
# #159316769 Setup the project folder structure

#### What does this PR do?
- This PR setup and refactor the project folder structure

#### Description of Task to be completed?
- Create a `server` folder for organizing server codebase
- Add the relevant folders to `server` folder
- Add the corresponding files and code
- Update the import references
- Refactor codebase to separate controllers definition from routes

#### How should this be manually tested?
- Checkout to `chore/159316769/setup-folder-structure` branch
- `npm test` to ensure the existing functionalities are working

#### Any background context you want to provide?
- This is the how the new folder structure looks like:
```
/server
- /configs
- /controllers
- /database
-- /migrations
-- /models
-- /seeders
- /routes
- /test
-- /routes
- /utils
-- /helpers
-- /middleware
-- /services
- index.js
package.json
```

Top level only:

```
/server
- /configs
- /controllers
- /database
- /routes
- /test
- /utils
- index.js
package.json
```
#### What are the relevant pivotal tracker stories?
#25
